components/src/dynamo/vllm/args.py

```
@@ -121,412 +81,241 @@ def parse_args() -> Config:
         return None
 
     # If prefix caching is not enabled, no events config needed
-    if not config.engine_args.enable_prefix_caching:
+    if not engine_config.enable_prefix_caching:
         logger.info("No kv_events_config required: prefix caching is disabled")
         return None
 
     # If user provided their own config, use that
-    if c := getattr(config.engine_args, "kv_events_config"):
-        # Warn user that enable_kv_cache_events probably should be True (user may have omitted it from JSON)
+    if c := getattr(engine_config, "kv_events_config"):
```

Ruff (0.15.0)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)